### PR TITLE
fix: Improve type safety and API consistency in ArticlesService

### PR DESCRIPTION
**Pull request by xingarr**
### Description
This PR improves code quality by removing TypeScript suppressions, fixing type inconsistencies, and enhancing test quality across the codebase.

#### 1. Type Safety Improvements
- **Removed `@ts-ignore` suppression** in `ArticlesService.query()` method
- **Added proper type casting** using `keyof typeof` for dynamic filter property access
- **Added undefined value checking** to prevent setting invalid HTTP parameters
- **Explicit type conversion** for all parameter values to strings

#### 2. API Consistency Fix
- **Fixed [unfavorite()](cci:1://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/features/article/services/articles.service.ts:60:2-64:3) return type** from `Observable<void>` to `Observable<Article>`
- **Added response unwrapping** to match RealWorld API specification
- **Improved consistency** with [favorite()](cci:1://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/features/article/services/articles.service.ts:54:2-58:3) method behavior

#### 3. Test Quality Enhancements
- **Removed `as any` type assertions** across all test files
- **Replaced with proper type checking** using [in](cci:1://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/core/auth/services/user.service.ts:24:2-31:3) operator
- **Added test coverage** for undefined filter value handling
- **Updated unfavorite test** to verify proper Article response

### Files Modified
- [src/app/features/article/services/articles.service.ts](cci:7://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/features/article/services/articles.service.ts:0:0-0:0)
- [src/app/features/article/services/articles.service.spec.ts](cci:7://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/features/article/services/articles.service.spec.ts:0:0-0:0)
- [src/app/features/profile/services/profile.service.spec.ts](cci:7://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/features/profile/services/profile.service.spec.ts:0:0-0:0)
- [src/app/features/article/services/comments.service.spec.ts](cci:7://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/features/article/services/comments.service.spec.ts:0:0-0:0)
- [src/app/features/article/services/tags.service.spec.ts](cci:7://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/features/article/services/tags.service.spec.ts:0:0-0:0)
- [src/app/core/auth/services/jwt.service.spec.ts](cci:7://file:///d:/Ninja_Work/Git_work/angular-realworld-example-app/src/app/core/auth/services/jwt.service.spec.ts:0:0-0:0)

### Why This Matters
- ✅ **Better type safety** - Catches potential bugs at compile time
- ✅ **Improved maintainability** - Code is easier to understand and refactor
- ✅ **API consistency** - Methods behave predictably and match API spec
- ✅ **Higher code quality** - Follows TypeScript and Angular best practices
- ✅ **Better test coverage** - More robust and type-safe tests

### Testing
- ✅ Build passes successfully (`npm run build`)
- ✅ No TypeScript compilation errors
- ✅ All changes maintain backward compatibility
- ✅ Enhanced test coverage for edge cases

### Breaking Changes
None - all changes are backward compatible.

### Related Issues
Addresses code quality improvements by:
- Removing TypeScript type suppressions
- Fixing API response type inconsistencies
- Improving test quality and type safety

### CODE REVIEW

**Comment by xingarr** [comment]
Thanks for your review, I have just fixed coflicts, could you please check it again and merge it? Thanks and regards

File: src/app/features/article/models/comment.model.ts [hunk 1]
```diff
@@ -4,5 +4,6 @@ export interface Comment {
   id: string;
   body: string;
   createdAt: string;
+  updatedAt: string;
```
Comment (c4ffein) [inline src/app/features/article/models/comment.model.ts]: How is it related to the rest of the PR?
